cookiecutter/cli.py

```diff
@@ -53,7 +52,9 @@ def print_version(context, param, value):
     '-v', '--verbose',
     is_flag=True, help='Print debug information', default=False
 )
-def main(template, no_input, checkout, verbose):
+@click.argument('template')
+@click.argument('extra_context', nargs=-1)
+def main(no_input, checkout, verbose, template, extra_context):
```

Comment:
Hmmm, it's been a while so can't remember thinking, but probably I moved the `template` argument after the options, because it seems more natural and POSIXy to have options before arguments and that's how the click documentation shows it ([example](http://click.pocoo.org/4/quickstart/#adding-parameters)).

Any concern about moving it (e.g.: any other tools that call the `main` function with positional arguments)? Easy enough to change if there is.